compiler/luci-interpreter/src/core/RuntimeGraph.cpp

```diff
@@ -89,16 +96,25 @@ void RuntimeGraph::TensorAllocPlan::deallocate(size_t kernel_index) const
   assert(_valid && kernel_index < _dealloc_plan.size());
   for (Tensor *tensor : _dealloc_plan[kernel_index])
   {
-    tensor->deallocate();
+    _memory_manager->release_memory(tensor);
   }
 }
 
-RuntimeGraph::RuntimeGraph(RuntimeModule *owning_module)
-  : _owning_module(owning_module), _tensor_alloc_plan(std::make_unique<TensorAllocPlan>())
+RuntimeGraph::RuntimeGraph(RuntimeModule *owning_module, IMemoryManager *memory_manager)
+  : _owning_module(owning_module),
+    _memory_manager(memory_manager),
+    _tensor_alloc_plan(std::make_unique<TensorAllocPlan>(memory_manager))
 {
 }
 
-RuntimeGraph::~RuntimeGraph() {}
+RuntimeGraph::~RuntimeGraph()
+{
+  for (int i = 0; i < _tensors.size(); i++)
```

Comment:
This line fails to compile:
`error: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<std::unique_ptr<luci_interpreter::Tensor> >::size_type’ {aka ‘long unsigned int’} [-Werror=sign-compare]`